src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,259 @@
+                + "FROM cars c "
+                + "JOIN manufacturers m ON c.manufacturer_id = m.id "
+                + "WHERE TRUE "
+                + "AND c.is_deleted = 0 ";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllCarsStatement = connection
+                        .prepareStatement(getAllCarsRequest,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            ResultSet resultSet = getAllCarsStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(parseCarFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all Cars", e);
+        }
+        for (Car car : cars) {
+            car.setManufacturer(getManufacturerForCar(car.getId()));
+            car.setDrivers(getDriversForCar(car.getId()));
+        }
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String createCarRequest =
+                "UPDATE cars "
+                + "SET manufacturer_id = ?, model = ? "
+                + "WHERE TRUE "
+                + "AND cars.id = ? "
+                + "AND is_deleted = 0 ";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(createCarRequest,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            statement.setLong(1, car.getManufacturer().getId());
+            statement.setString(2, car.getModel());
+            statement.setLong(3, car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update Car: " + car, e);
+        }
+        deleteRelationWithCar(car);
+        addDriversToCarsDriversTable(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String createCarRequest =
+                "UPDATE cars "
+                + "SET is_deleted = 1 "
+                + "WHERE TRUE "
+                + "AND cars.id = ? "
+                + "AND is_deleted = 0 ";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(createCarRequest,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            statement.setLong(1, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete Car with id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String getAllCarsRequest =
+                "SELECT d.id AS driver_id, d.name AS driver_name, d.license_number, "
+                + "c.id AS car_id, c.manufacturer_id, c.model "
```

Comment:
why do we need to get info about drivers here? And lets get all info about manufacturer